### PR TITLE
Add sold_after and sold_before in ranking API; test cases refactor

### PR DESCRIPTION
**Pull request by nnkken**
In this PR, `Before` and `After` in ranking API refers to sold time instead of creation time.
The original creation time semantic are moved to `CreatedBefore` and `CreatedAfter` fields.

Also, test cases are refactored (again) so now database data for test cases are created locally in each test case instead of shared globally.
This allows creating test cases easier.

Also added address migration, which includes (a) converting addresses before writing into database; (b) a parallel migration (which takes half a day to finish) for existing data.
Running that is not urgent nor necessary, since most of the queries are querying a set of addresses containing both prefixes. However, some counts may be affected (e.g. duplicate entries in nft count).

Also fixed ~tones~ some of the queries.

### CODE REVIEW

File: rest/nft.go [hunk 1]
```diff
@@ -5,6 +5,10 @@ import (
 	"github.com/likecoin/likecoin-chain-tx-indexer/db"
 )
 
+var (
+	DEFAULT_API_ADDRESSES = []string{"like17m4vwrnhjmd20uu7tst7nv0kap6ee7js69jfrs"}
```
Comment (williamchong) [inline rest/nft.go:9]: Can this be overide by a env/config? Useful when deploying testnet

File: db/schema/parallel.go [hunk 2]
```diff
@@ -88,6 +97,258 @@ func MigrationSetupIscnVersionTable(conn *pgxpool.Conn, batchSize uint64) error
...
+			if err != nil {
+				logger.L.Warnw(
+					"Cannot convert ISCN owner address",
+					"pkey_id", pkeyId,
+					"owner", owners[i],
+					"error", err,
+				)
+				continue
+			}
+			batch.Queue(
```
Comment (williamchong) [inline db/schema/parallel.go:146]: Can we not update if `convertedOwner === owners[i]`?

**Review by williamchong — CHANGES_REQUESTED** [review]
Still haven't read nft_test.go yet, will read tmr

**Review by williamchong — APPROVED** [review]

**Comment by williamchong** [comment]
image
Probably should check not empty string for address migration too

**Comment by nnkken** [comment]
> image
> 
> Probably should check not empty string for address migration too

Conversion error is expected, so it's a warning rather than an error

**Pull request merged**